src/util/compat.test.js

```diff
@@ -0,0 +1,169 @@
+// @flow
+
+import {toCompat, fromCompat} from "./compat";
+import type {Compatible} from "./compat";
+
+describe("compat utilities", () => {
+  type FooV1 = {foo: number};
+  type FooV2 = {bar: {foo: number}};
+  const type = "Foo Plugin's Experiment";
+  const v1 = "v1";
+  const v2 = "v2";
+  function fooHandler(x: FooV1): FooV2 {
+    return {bar: x};
+  }
+  const example = () => {
+    const dataV1: FooV1 = {foo: 1};
+    const dataV2: FooV2 = {bar: {foo: 1}};
+    const compatV1 = toCompat(type, v1, dataV1);
+    const compatV2 = toCompat(type, v2, dataV2);
+    return {dataV1, dataV2, compatV1, compatV2};
+  };
+  it("toCompat adds a compat field", () => {
+    const {dataV1, compatV1} = example();
+    expect(dataV1).toEqual({foo: 1}); // calling compat caused no mutation
+    const {type: actualType, version: actualVersion} = (compatV1: any)[0];
```

Comment:
Doesn't look like this test provides much value—the others are sufficient—and it deliberately breaks the encapsulation, so let's get rid of it.

Reply:
Done